### PR TITLE
gh-102 make Safe not stored properties as transient; refactor SafeSet…

### LINKED ISSUE #102: Try out Transient CoreData properties
instead of manually added properties.

### PR DESCRIPTION
**Pull request by sche**
closes #102

*What I checked:*
- Adding Safe by ENS
- Safe Settings
 -- Changing Safe name
 -- Switching Safes
- Removing Safe
- No Safes views

### CODE REVIEW

File: Multisig/UI/Settings/Safe Settings/LoadableSafeSettingsViewModel.swift [hunk 1]
```diff
@@ -48,24 +64,32 @@ class LoadableSafeSettingsViewModel: ObservableObject {
...
+extension Safe {
     func update(from safeInfo: SafeStatusRequest.Response) {
-        objectWillChange.send()
-        threshold = safeInfo.threshold
+        threshold = Int32(safeInfo.threshold)
         owners = safeInfo.owners
         masterCopy = safeInfo.masterCopy
         version = safeInfo.version
-        nonce = safeInfo.nonce
+        nonce = Int32(safeInfo.nonce)
```
Comment (DmitryBespalov) [inline Multisig/UI/Settings/Safe Settings/LoadableSafeSettingsViewModel.swift:91]: we really need to have the response validation, as the values can potentially overflow the Int32. We have this as a different tech debt ticket already.

**Review by DmitryBespalov — COMMENTED** [review]
I need to spend some more time to understand the fetched results controller

File: Multisig/Logic/Models/Safe.swift [hunk 2]
```diff
@@ -21,7 +21,30 @@ extension Safe: Identifiable {
 
     var displayName: String { name.flatMap { $0.isEmpty ? nil : $0 } ?? "Untitled Safe" }
 
-    var displayENSName: String { "" }
+    var displayENSName: String {
+        // TODO: https://github.com/gnosis/safe-ios/issues/202
+        ""
+    }
+
+    var owners: [String]? {
```
[the remark this answers is not in the document]
Comment (sche) [reply Multisig/Logic/Models/Safe.swift:29]: Yes, a good idea. I will try to make it with a property wrapper

[the remark this answers is not in the document]
Comment (sche) [reply Multisig/Logic/Models/Safe.swift:29]: "Extensions must not contain stored properties"
So property wrapper does not benefit in this case.

**Comment by sche** [comment]
The mentioned bug is not always reproducible. We checked it on several simulators, and on one computer it reproduces under some simulator, on another it does not. We checked in iPhone X and iPhone 6s devices, and there it does not reproduce.
Steps:
1. Add a single safe with the name "alice.eth"
2. Go to settings and remove the safe
=> Crash : SIGABRT "2020-06-08 11:52:08.839986+0200 Multisig[86115:14577131] precondition failure: unknown attribute: 4294967295"

So we decided to postpone merging of this PR. It needs proper testing on real devices.

**Review by MouazAlzahabi — APPROVED** [review]

**Comment by DmitryBespalov** [comment]
temporary closing as this is too old

**Pull request closed without merging**